### PR TITLE
Dynamic dags update

### PR DESCRIPTION
**Pull request by kentdanas**
@jwitz @vparekh94 would love your review on this one. For context, this is an existing guide that was due for an update. I made quite a few changes to the existing content (added an overview, supporting github repo, more context around the examples, pros and cons of the methods presented), and also added a few new methods, plus a section on scalability. Let me know if your thoughts :)

### CODE REVIEW

**Review by jwitz — CHANGES_REQUESTED** [review]
Great updates here @kentdanas, mostly minor wording changes and some small questions throughout. I'll take another pass after you get a chance to read through my suggestions.

File: guides/dynamically-generating-dags.md [hunk 1]
```diff
@@ -6,53 +6,43 @@ slug: "dynamically-generating-dags"
...
-```
+- It's simple and easy to implement.
+- It can accommodate input parameters from many different sources (see a few examples below).
+- Adding DAGs is nearly instantaneous since it only requires changing the input parameters.
 
-![title](https://assets.astronomer.io/website/img/guides/hello_world.png)
+But, there are also drawbacks, including:
 
-## Add DAGs dynamically based on input parameter
+- Since a DAG file isn't actually being created, your visibility into the code behind any specific DAG is limited.
```
[the remark this answers is not in the document]
Comment (kentdanas) [reply guides/dynamically-generating-dags.md:32]: The `Code` tab in Airflow will only show you the code that was parsed to create that DAG, so in this case, only the script that dynamically created it. With this method you aren't actually creating the DAG code anywhere that is persisted.

File: guides/dynamically-generating-dags.md [hunk 2]
```diff
@@ -6,53 +6,43 @@ slug: "dynamically-generating-dags"
...
+- Since this method requires a Python file in the `DAG_FOLDER` to dynamically generate the DAGs, the code will be executed on every scheduler heartbeat. This can cause performance issues if the total number of DAGs is large, or if the code is connecting to an external system such as a database. For more on this see the Scalability section below.
 
-Multiple DAGs can be registered from the same file, but to improve maintainability and avoid namespace conflicts, it is advisable to keep one file per one unique DAG. However, if multiple unique DAGs are required with the same base code, it is possible to create these DAGs dynamically based on any number of configuration parameters.
+Below, we show a few different examples of how to implement this pattern using different input parameters for creating the DAGs.
 
-A common use case for this is when pulling data from multiple APIs or database that have a similar incoming structure, require similar transform logic, and need to be loaded according to a similar pattern.
 
-### Create_DAG method
+### Create_DAG Method
```
[the remark this answers is not in the document]
Comment (kentdanas) [reply guides/dynamically-generating-dags.md:38]: Yeah I struggled with how to present this. Essentially this section defines the function that dynamically generates DAGs, and the sections below show different examples of how to call that function based on different input parameters. I'm fine adding `Example` to the title, but also open to other ideas if you have any.

File: guides/dynamically-generating-dags.md [hunk 3]
```diff
@@ -77,14 +67,12 @@ def create_dag(dag_id,
     return dag
 ```
 
-We can then set a simple loop (`range(1, 10)`) to generate these unique parameters and pass them to the global scope, thereby registering them as valid DAGs to the Airflow scheduler.
+With the code above, the input parameters can come from any source that the Python script can access. For this first example, we set a simple loop (`range(1, 4)`) to generate these unique parameters and pass them to the global scope, thereby registering them as valid DAGs to the Airflow scheduler. That code looks like this:
```
[the remark this answers is not in the document]
Comment (kentdanas) [reply guides/dynamically-generating-dags.md:70]: Yes

File: guides/dynamically-generating-dags.md [hunk 4]
```diff
@@ -127,23 +113,23 @@ for n in range(1, 10):
                                   default_args)
 ```
 
+And if we look at the Airflow UI we can see the DAGs have been created:
+
 ![title](https://assets.astronomer.io/website/img/guides/hello_world_1_thru_10.png)
```
[the remark this answers is not in the document]
Comment (kentdanas) [reply guides/dynamically-generating-dags.md:118]: Oh thanks, I didn't even notice that!

**Comment by kentdanas** [comment]
> Great updates here @kentdanas, mostly minor wording changes and some small questions throughout. I'll take another pass after you get a chance to read through my suggestions.

Thanks @jwitz! I accepted your suggestions and I think answered all your questions. Let me know if you want to discuss anything else.

File: guides/dynamically-generating-dags.md [hunk 5]
```diff
@@ -162,58 +148,51 @@ def create_dag(dag_id,
...
     globals()[dag_id] = create_dag(dag_id,
                                   schedule,
                                   dag_number,
                                   default_args)
 
-
 ```
 
-If we look at the scheduler logs, we can see this variable is being pulled into the DAG and added an additional DAG to the DagBag based on the value.
+If we look at the scheduler logs, we can see this variable was pulled into the DAG and, and 15 DAGs were added to the DagBag based on the its value.
```
Comment (jwitz) [inline guides/dynamically-generating-dags.md:175]:
```suggestion
If we look at the scheduler logs, we can see this variable was pulled into the DAG and, and 15 DAGs were added to the DagBag based on its value.
```

**Review by jwitz — APPROVED** [review]
Found one minor typo, but otherwise looks good to me!

**Review by vparekh94 — APPROVED** [review]

**Pull request merged**